doc/source/user/basics.types.rst

```diff
@@ -12,6 +12,138 @@ Array types and conversions between types
 NumPy supports a much greater variety of numerical types than Python does.
 This section shows which are available, and how to modify an array's data-type.
 
+NumPy numerical types are instances of `numpy.dtype` (data-type) objects, each
+having unique characteristics.  Once you have imported NumPy using ``import
+numpy as np`` you can create arrays with a specified dtype using the scalar
+types in the numpy top-level API, e.g. `numpy.bool`, `numpy.float32`, etc.
+
+These scalar types as arguments to the dtype keyword that many numpy functions
+or methods accept. For example::
+
+    >>> z = np.arange(3, dtype=np.uint8)
+    >>> z
+    array([0, 1, 2], dtype=uint8)
+
+Array types can also be referred to by character codes, for example::
+
+  >>> np.array([1, 2, 3], dtype='f')
+  array([1.,  2.,  3.], dtype=float32)
+  >>> np.array([1, 2, 3], dtype='d')
+  array([1.,  2.,  3.], dtype=float64)
+
+See :ref:`arrays.dtypes.constructing` for more information about specifying and
+constructing data type objects, including how to specify parameters like the
+byte order.
+
+To convert the type of an array, use the .astype() method (preferred) or
```

Comment:
I'd still stop the sentence after "method" - no need for introducing the "type itself as a function"